### DISCUSSION TITLE
A prescription for Cortex-M application builds using gcc+make

### DISCUSSION THREAD

**Discussion by tobermory** [ARM-software/CMSIS_5#1539] [Show and tell]
I advertised a small library for doing structured fault handling on Cortex-M cpus on Github, in response to a request for info on inferring function call stacks in fault conditions.

I then realized that its Makefile details the steps required to build arbitrary applications for Cortex-M using just gcc, make. Add say JLink Commander as flash tool and you have a complete build/install toolset.

https://github.com/tobermory/faultHandling-cortex-m.git

**Comment by JonatanAntoni** [comment]
Hi @tobermory,

Thanks for advertising this. Please note that the "standard" way of building (or better composing) bare metal applications for Arm Cortex processors is using [Open-CMSIS-Pack](https://github.com/Open-CMSIS-Pack). If you like, you could release your analysis component as a pack and let us register it to our pack index. This would give easy access to your software component to our ecosystem user base.

Cheers,
Jonatan

**Comment by tobermory** [comment]
Thank you for the info.

I tried to find some examples of published packs by individuals, but can only find vendor ones here:

https://developer.arm.com/tools-and-software/embedded/cmsis/cmsis-packs

That link I found in a memfault blog describing PACKs.

I am trying to discover which tools ingest .pack files?  Is Keil uVision the only one?

**Reply by JonatanAntoni** [reply]
This page should give you an extensive list of available packs from over 60 silicon and software partners. Not sure why it appears empty for you. The list is dynamically loaded which may take a while. There seems not to be a good notification when loading is in progress.

Today, the Arm IDEs ArmDS, Keil MDK (uVision) and Keil Studio support packs. Furthermore IAR Embedded Workbench, ST Cube IDE and NXP MCUXpresso have support as well.

**Comment by tobermory** [comment]
Can you point me at specific instructions for taking some arbitrary set of .h, .c, .S and perhaps Makefiles and folding them into the .pack format? Thanks.

**Reply by JonatanAntoni** [reply]
Hi @tobermory,

The latest documentation for Open-CMSIS-Pack can be found at https://open-cmsis-pack.github.io/Open-CMSIS-Pack-Spec/main/html/index.html

You can find some tutorials about creating packs, description how to publish a pack and a detailed description of the data format used.

Please let me know if you have any further questions. Any enhancement proposal to the documentation is highly appreciated.

Cheers,
Jonatan
